cargo-quickinstall/src/lib.rs

```diff
@@ -0,0 +1,266 @@
+            println!("Falling back to `cargo install`.");
+
+            let status = prepare_cargo_install_cmd(details).status()?;
+
+            if status.success() {
+                Ok(())
+            } else {
+                Err(InstallError::CargoInstallFailed)
+            }
+        }
+        Err(err) => Err(err),
+    }
+}
+
+pub fn get_latest_version(crate_name: &str) -> Result<String, InstallError> {
+    let url = format!("https://crates.io/api/v1/crates/{}", crate_name);
+
+    let parsed = curl_json(&url).map_err(|e| {
+        if e.is_curl_404() {
+            InstallError::CrateDoesNotExist {
+                crate_name: crate_name.to_string(),
+            }
+        } else {
+            e
+        }
+    })?;
+    Ok(parsed["versions"][0]["num"].clone().try_into().unwrap())
+}
+
+pub fn get_target_triple() -> Result<String, InstallError> {
+    // Credit to https://stackoverflow.com/a/63866386
+    let output = std::process::Command::new("rustc")
+        .arg("--version")
+        .arg("--verbose")
+        .output()?;
+    for line in String::from_utf8(output.stdout).unwrap().lines() {
+        if let Some(target) = line.strip_prefix("host: ") {
+            return Ok(target.to_string());
+        }
+    }
+    Err(CommandFailed {
+        command: "rustc --version --verbose".to_string(),
+        stdout: "".to_string(),
+        stderr: String::from_utf8(output.stderr).unwrap(),
+    }
+    .into())
+}
+
+pub fn report_stats_in_background(details: &CrateDetails) -> std::thread::JoinHandle<()> {
+    let tarball_name = format!(
+        "{}-{}-{}.tar.gz",
+        details.crate_name, details.version, details.target
+    );
+
+    let stats_url = format!(
+        "https://warehouse-clerk-tmp.vercel.app/api/crate/{}",
+        tarball_name
+    );
+    std::thread::spawn(move || {
+        // warehouse-clerk is known to return 404. This is fine. We only use it for stats gathering.
+        curl_head(&stats_url).unwrap_or_default();
+    })
+}
+
+pub fn do_dry_run(crate_details: &CrateDetails) -> String {
+    let result: String;
+
+    let crate_download_url = format!(
+        "https://github.com/alsuren/cargo-quickinstall/releases/download/\
+                 {crate_name}-{version}-{target}/{crate_name}-{version}-{target}.tar.gz",
+        crate_name = crate_details.crate_name,
+        version = crate_details.version,
+        target = crate_details.target
+    );
+    if curl_head(&crate_download_url).is_ok() {
+        let cargo_bin_dir = home::cargo_home().unwrap().join("bin");
+        let cargo_bin_dir_str = cargo_bin_dir.to_str().unwrap();
+        result = format!(
+            "{curl_cmd:?} | {untar_cmd:?}",
+            curl_cmd = prepare_curl_bytes_cmd(&crate_download_url),
+            untar_cmd = prepare_untar_cmd(cargo_bin_dir_str)
+        );
+    } else {
+        let cargo_install_cmd = prepare_cargo_install_cmd(crate_details);
+        result = format!("{:?}", cargo_install_cmd);
+    }
+
+    result
```

Comment:
```suggestion
```

Reply:
`if` statements are expressions in rust, so you can return directly from them. There's no need for the `result` variable.